Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -66,12 +66,48 @@ static ArrayBuffer* array_buffer_object_from(VM& vm, GlobalObject& global_object
     return static_cast<ArrayBuffer*>(&this_object);
 }
 
+// 25.1.5.3 ArrayBuffer.prototype.slice, https://tc39.es/ecma262/#sec-arraybuffer.prototype.slice
 JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::slice)
 {
+    const auto start = vm.argument(0);
+    const auto end = vm.argument(1);
+
     auto array_buffer_object = array_buffer_object_from(vm, global_object);
     if (!array_buffer_object)
         return {};
-    TODO();
+
+    // FIXME: Check for shared buffer
+    // FIXME: Check for detached buffer
+
+    const auto len = array_buffer_object->byte_length();
+
+    const auto relative_start = start.is_negative_infinity()
+        ? 0
+        : start.to_integer_or_infinity(global_object);
```

Comment:
Missing exception check, i.e.

```suggestion
        : start.to_integer_or_infinity(global_object);
    if (vm.exception())
        return {};
```